engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -280,46 +280,56 @@ namespace dmGameSystem
         return GetMaterialResource(component, resource, index)->m_Material;
     }
 
-    TextureResource* GetTextureResource(const ModelComponent* component, uint32_t material_index, uint32_t texture_unit)
+    static TextureResource* GetTextureFromSamplerNameHash(const MaterialInfo* material_info, const MaterialResource* material, uint32_t material_texture_index, dmhash_t sampler_name_hash)
```

Comment:
We need to think of the sampler names, to figure out their positions in the render objects' texture array.

Here we ask using sampler name first, then using the material texture index (as we're currently looping over the material's samplers, we can use that index directly)